src/components/main-view/main-view.jsx

```diff
@@ -3,23 +3,55 @@ import { MovieCard } from "../movie-card/movie-card";
 import { MovieView } from "../movie-view/movie-view";
 import { LoginView } from "../login-view/login-view";
 import { SignUpView } from "../sign-up-view/sign-up-view";
+import { NavigationBar } from "../navigation-bar/navigation-bar";
 import "./main-view.scss";
 import Row from "react-bootstrap/Row";
 import Col from "react-bootstrap/Col";
-import Button from "react-bootstrap/Button";
-import { Link } from "react-router-dom";
 
 import { BrowserRouter, Routes, Route, Navigate } from "react-router-dom";
+import { ProfileView } from "../profile-view/profile-view";
 
 export const MainView = () => {
   const [movies, setMovies] = useState([]);
 
-  let [selectedMovie, setSelectedMovie] = useState(null);
   const storedUser = JSON.parse(localStorage.getItem("user"));
   const storedToken = localStorage.getItem("token");
   const [user, setUser] = useState(storedUser ? storedUser : null);
   const [token, setToken] = useState(storedToken ? storedToken : null);
-  let [signUp, setSignUp] = useState(null);
+  const [favoriteMovies, setFavoriteMovies] = useState([]);
+
+  const handleOnLoggedOut = () => {
+    setUser(null);
+    setToken(null);
+    localStorage.clear();
+  };
+
+  const addToFavoritesList = (movieToAdd) => {
+    let updatedUser = user;
+    updatedUser.favoriteMovies.push(movieToAdd);
+    setUser(updatedUser);
+    generateNewFavoriteMovies();
+  };
+
+  const removeFromFavoritesList = (movieToRemove) => {
+    let updatedUser = user;
+
+    for (i = 0; i < updatedUser.favoriteMovies.length; i++) {
+      if (updatedUser.favoriteMovies[i] === movieToRemove) {
+        delete updatedUser.favoriteMovies[i];
+        break;
+      }
+    }
+    setUser(updatedUser);
+    generateNewFavoriteMovies();
+  };
+
+  const generateNewFavoriteMovies = () => {
+    let usersFavoriteMovies = movies.filter((m) => {
+      return user.favoriteMovies.includes(m.id);
```

Comment:
Hi Frank,
I am sorry maybe I understood something wrong.
The function generateNewFavoriteMovies gets called after a movie has been added or removed to/from the favorites array of a user.(which consists only of a string)
in lines 50-52 I create a new temporary favorite movies array(with all the neccessay data to render the movies) 
The state is changed in line 53 where I use **setFavoriteMovies** to set the value of favoriteMovies by using the temporary list.

53:   setFavoriteMovies(usersFavoriteMovies);

Could you please share more thoughts because right now I do not understand what exactly I did wrong.